### PR TITLE
8282764: AArch64: compiler/vectorapi/reshape/TestVectorCastNeon.java failed with incorrect result

### PR DESCRIPTION
**Pull request by nsjian**
Vector API long to float conversion operation on NEON converts vector of 2 longs to 2 floats. In current implementation, we convert 2 longs to 2 doubles first and then converts 2 doubles to 2 floats. However, this two-steps conversion may have two roundings, while the expected behavior, conversion from long to float directly, has only one rounding. This will result in inconsistent result. E.g. for the failure test case:

```
jshell> long l = 0x561a524000000001L;
l ==> 6204361871487664129

jshell> float l2f = (float)l;
l2f ==> 6.2043621E18

jshell> float l2d2f = (float)((double)l);
l2d2f ==> 6.2043616E18
```

Since we don't have NEON instruction to support vector long to float conversion, we fix the codegen by doing it in scalar operation one element by one element.

TEST: vector api jtreg tests passed.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8282764](https://bugs.openjdk.java.net/browse/JDK-8282764): AArch64: compiler/vectorapi/reshape/TestVectorCastNeon.java failed with incorrect result

### Reviewers
 * [Paul Sandoz](https://openjdk.java.net/census#psandoz) (@PaulSandoz - **Reviewer**)
 * [Nick Gasson](https://openjdk.java.net/census#ngasson) (@nick-arm - **Reviewer**)
 * [Eric Liu](https://openjdk.java.net/census#eliu) - Author ⚠️ Added manually

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7850/head:pull/7850` \
`$ git checkout pull/7850`

Update a local copy of the PR: \
`$ git checkout pull/7850` \
`$ git pull https://git.openjdk.java.net/jdk pull/7850/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7850`

View PR using the GUI difftool: \
`$ git pr show -t 7850`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7850.diff">https://git.openjdk.java.net/jdk/pull/7850.diff</a>

</details>

### CODE REVIEW

**Comment by theRealELiu** [comment]
Looks good to me (not a reviewer).

**Comment by nsjian** [comment]
Thank you @theRealELiu !

**Comment by nsjian** [comment]
> Looks good to me (not a reviewer).

Thank you @theRealELiu !

File: test/hotspot/jtreg/compiler/vectorapi/VectorCastShape128Test.java [hunk 1]
```diff
@@ -114,6 +114,7 @@
         0,
         Long.MAX_VALUE,
         Long.MIN_VALUE,
+        0x561a524000000001L,
```
Comment (PaulSandoz) [inline test/hotspot/jtreg/compiler/vectorapi/VectorCastShape128Test.java:117]: A comment would be useful as to why this constant is important

**Review by PaulSandoz — COMMENTED** [review]
Thanks for updating the copyright year. Java changes look good. Just one comment related to an additional value added for testing.

**Comment by nsjian** [comment]
> Thanks for updating the copyright year. Java changes look good. Just one comment related to an additional value added for testing.

Thank you for the review @PaulSandoz ! Code comment added.

**Review by PaulSandoz — APPROVED** [review]
Java changes are good.

**Review by nick-arm — APPROVED** [review]
cpu/aarch64 changes look OK to me and the Windows x64 failure is some unrelated environment setup problem.

**Comment by nsjian** [comment]
Thanks for the review! @nick-arm 

/reviewer credit @theRealELiu

**Pull request closed without merging**